azure-pipelines.yml

```diff
@@ -9,8 +9,8 @@ variables:
   DO_BUILD: 'yes'
   DO_TEST: 'yes'
   DO_CLEAN: 'no'
-  CLANG14_IMAGENAME: 'axom/tpls:clang-14_09-23-24_17h-07m'
-  GCC13_IMAGENAME: 'axom/tpls:gcc-13_09-23-24_17h-07m'
+  CLANG14_IMAGENAME: 'axom/tpls:clang-14_10-10-24_01h-31m'
```

Comment:
Is this change required?

Reply:
Yes.  The new image has Umpire shared libs.  That's the key to the bug fix.

Reply:
@white238 @kennyweiss are you OK with switching to Umpire shared libs for azure testing?

Reply:
We (@kennyweiss , @gunney1 , @davidbeckingsale ) met about this last night. Basically a static umpire with a shared axom creates multiple singletons in Umpire. We need to not allow this to happen in our Spack recipe as well but this seems to be the only way around this.

Reply:
Do you mean add a constraint in the Axom Spack package that static Umpire conflicts with shared Axom?

Reply:
yes and/or a conflict

Reply:
Yes -- the three of us had a meeting yesterday w/ @davidbeckingsale and concluded that the static umpire combined w/ a shared axom in the docker configuration was leading to more than one umpire instance. 

There might be a solution to this w/ improved linking orders and/or linker flags, but it will be difficult to track down in the short term, and is not necessary for @gunney1's actual deliverable (i.e. the CI is testing a different configuration than the one we actually need). @gunney1 is planning to create an issue summarizing that discussion so we can look into it in the future. 

It's also worth noting that we've encountered similar issues in the past when mixing libraries w/ singletons with upstream shared or Fortran libraries (e.g. #865 and #270).